src/Model/Resolver/Menu.php

```diff
@@ -84,43 +75,49 @@ public function resolve(
         ResolveInfo $info,
         array $value = null,
         array $args = null
-    ): Value {
-        $result = function () {
-            return null;
-        };
-
-        if (isset($args['identifier'])) {
-            $menu = $this->menuCollectionFactory->create();
-
-            /** Updated with identifier filtering */
-            $menu->addFieldToFilter('identifier', $args['identifier'])->load();
+    )
+    {
+        $identifier = $args['identifier'];
 
-            $menuData = $menu->getFirstItem()->getData();
+        $menu = $this->menuFactory
+            ->create()
+            ->load($identifier);
```

Comment:
Do not load on Model, use Resource model, ie:
$menu = $this->menuFactory->create();
$this->resourceModel->load($menu, $identifier)